packages/core/src/tools/mcp-client.ts

```diff
@@ -301,6 +307,44 @@ export async function discoverTools(
   }
 }
 
+const PromptListResponseSchema = z.object({
+  prompts: z.array(
+    z.object({
+      name: z.string(),
+      description: z.string().optional(),
+    }),
+  ),
+});
+
+/**
+ * Discovers and logs prompts from a connected MCP client.
+ * It retrieves prompt declarations from the client and logs their names.
+ *
+ * @param mcpServerName The name of the MCP server.
+ * @param mcpClient The active MCP client instance.
+ */
+export async function discoverPrompts(
+  mcpServerName: string,
+  mcpClient: Client,
+): Promise<void> {
+  try {
+    const response = await mcpClient.request(
+      { method: 'prompts/list', params: {} },
```

Comment:
Should double check if the Anthropic SDK has a set of MCP methods (as static strings) for this